sktime/forecasting/lagllama.py

```diff
@@ -0,0 +1,348 @@
+"""Implements an adapter for the LagLlama estimator for intergration into sktime."""
+
+__author__ = ["shlok191"]
+
+import subprocess
+
+from sktime.forecasting.base import _BaseGlobalForecaster
+
+
+class LagLlamaForecaster(_BaseGlobalForecaster):
+    """Base class that interfaces the LagLlama forecaster.
+
+    Parameters
+    ----------
+    weights_url : str, optional (default="time-series-foundation-models/Lag-Llama")'
+        The URL of the weights on hugging face for the LagLlama estimator to fetch.
+
+    device : str, optional (default="cpu")
+        Specifies the device on which to load the model.
+
+    context_length: int, optional (default=32)
+        The number of prior timestep data entries provided.
+
+    prediction_length: int, optional (default=100)
+        The length of future of timesteps to predict
+        (synonymous to forecast horizon).
+
+    num_samples: int, optional (default=10)
+        Number of sample paths desired for evaluation.
+
+    batch_size: int, optional (default=32)
+        The number of batches to train for in parallel.
+
+    nonnegative_pred_samples: bool, optional (default=False)
+        If True, ensures all predicted samples are passed
+        through ReLU,and are thus positive or 0.
+
+    lr: float, optional (default=5e-5)
+        The learning rate of the model.
+
+    trainer_kwargs: dict, optional (default={"num_epochs": 50})
+        The arguments to pass to the GluonTS trainer.
+    """
+
+    _tags = {
+        "y_inner_mtype": "gluonts_PandasDataset_panel",
+        "X_inner_mtype": "gluonts_PandasDataset_panel",
+        "scitype:y": "both",
+        "ignores-exogeneous-X": True,
+        "requires-fh-in-fit": True,
+        "X-y-must-have-same-index": True,
+        "enforce_index_type": None,
+        "handles-missing-data": False,
+        "capability:insample": True,
+        "capability:pred_int": True,
+        "capability:pred_int:insample": True,
+        "authors": ["shlok191"],
+        "maintainers": ["shlok191"],
+        "python_version": None,
+        "python_dependencies": ["gluonts", "huggingface_hub", "lag_llama"],
+    }
+
+    def __init__(
+        self,
+        weights_url=None,
+        device=None,
+        context_length=None,
+        prediction_length=None,
+        num_samples=None,
+        batch_size=None,
+        nonnegative_pred_samples=None,
+        lr=None,
+        trainer_kwargs=None,
+    ):
+        # Initializing parent class
+        super().__init__()
+
+        import torch
+        from lag_llama.gluon.estimator import LagLlamaEstimator
+
+        # Defining private variable values
+        self.weights_url = weights_url
+        self.weights_url_ = (
+            "time-series-foundation-models/Lag-Llama"
+            if not weights_url
+            else weights_url
+        )
+
+        self.device = device
+        self.device_ = torch.device("cpu") if not device else device
```

Comment:
```suggestion
        self.device_ = torch.device("cpu") if not device else torch.device(device)
```
Does it have to be something like this?

Reply:
Added it in a new commit as well